File: local_examples/set_and_get/ruby/set_get.rb
# EXAMPLE: set_and_get
# HIDE_START
require 'redis'

r = Redis.new
# HIDE_END

# REMOVE_START
def assert_equal(expected, actual)
  raise "Expected #{expected.inspect}, got #{actual.inspect}" unless actual == expected
end
r.del('bike:1')
# REMOVE_END

res1 = r.set('bike:1', 'Process 134')
puts res1 # >>> OK

res2 = r.get('bike:1')
puts res2 # >>> Process 134

# REMOVE_START
assert_equal('OK', res1)
assert_equal('Process 134', res2)
r.del('bike:1')
r.close
# REMOVE_END

File: local_examples/set_and_get/ioredis/set-get.js
// EXAMPLE: set_and_get

// HIDE_START
import assert from 'node:assert';
import { Redis } from 'ioredis';

const redis = new Redis();
// HIDE_END

// REMOVE_START
await redis.del('bike:1');
// REMOVE_END

const res1 = await redis.set('bike:1', 'Process 134');
console.log(res1); // >>> OK

const res2 = await redis.get('bike:1');
console.log(res2); // >>> Process 134

// REMOVE_START
assert.equal(res1, 'OK');
assert.equal(res2, 'Process 134');
await redis.del('bike:1');
// REMOVE_END

// HIDE_START
redis.disconnect();
// HIDE_END
